File: codingsample/202008/20200820/Island_of_Knowledge_1_areEquallyStrong.java
package Island_of_Knowledge;

public class Island_of_Knowledge_1_areEquallyStrong {
	static boolean areEquallyStrong(int yourLeft, int yourRight, int friendsLeft, int friendsRight) {
		return Math.max(yourLeft, yourRight)==Math.max(friendsLeft, friendsRight)&&Math.min(yourLeft, yourRight)==Math.min(friendsLeft, friendsRight);
	}
	public static void main(String[] args) { 
		int a1=15,a2=10,b1=10,b2=15;
		System.out.println(areEquallyStrong(a1,a2,b1,b2));
	}

}

File: codingsample/202008/20200820/Island_of_Knowledge_1_areEquallyStrong.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

bool areEquallyStrong(int yourLeft, int yourRight, int friendsLeft, int friendsRight) {
	if(yourLeft!=friendsLeft&&yourLeft!=friendsRight) return false;
	else if(yourRight!=friendsLeft&&yourRight!=friendsRight) return false;
	return true;
}

int main() {
	int a1 = 10, a2 = 20, b1=20, b2=11;
	if(areEquallyStrong(a1,a2,b1,b2)) printf("True\n");
	else printf("False\n");
}


